### ISSUE TITLE
Wrong reference to date field in child table

### ISSUE THREAD

**Issue by Nube2021** [corcel/corcel#218]
Hi,

I have two tables and one pivot table connecting them together.
- My first table is "users". The table is in my WordPress database. The model for this table is an extension of Corcel ("class User extends Corcel")
- My second table is "articles". The table is in my Laravel database. The model is the standard Laravel Model ("class Article extends Model")
- the pivot table is "article_user". The table is in my Laravel database. I have no model for this table.

In my User Model, I have the following relationship to the Article model:

`public function articles ()
    {
        return $this->belongsToMany('App\Article')->withTimestamps();
    }`

I added the "->withTimestamps()" as described in the Laravel docs in order to update/insert the date fields when a relationship is created/updated.

When the User Model extends the standard Laravel model, everythins works find. When the User Model extends Corsel, I get this error message:

Column not found: 1054 Unknown column 'article_user.user_registered' in 'field list' (SQL: select `articles`.*, `article_user`.`user_id` as `pivot_user_id`, `article_user`.`article_id` as `pivot_article_id`, `article_user`.`user_registered` as `pivot_user_registered`, `article_user`.``as`pivot_`from`articles`inner join`article_user`on`articles`.`id`=`article_user`.`article_id`where`article_user`.`user_id`= 1 and`articles`.`id`= 3 and`articles`.`deleted_at` is null limit 1)

**Comment by jgrossi** [comment]
Hi @Nube2021. This seems to be a connection problem. I suppose you have different connections for both Laravel and WordPress right? So for `User` model you set 
```
protected $connection = 'corcel'; // for example
```
and for the `Article` model just the `default` connection Laravel provides.

Are you doing this?

The problem in this SQL query is with `user_registered` column for `article_user` table. Does it exist? You use only `user_id = 1` in this query that is related to `User` model.

Is there anything missing?

**Comment by Nube2021** [comment]
Hi jgrossi,

you are right, User is corcel connection, Article the default connection.


The problem here is that when you set `const CREATED_AT = 'user_registered';` in jgrossi/corcel/src/User.php, it also affects what Laravel expects as timestamp fields in the pivot table. 
This means that Laravel expects that the timestamp creation date field in the pivot table is "user_registered".

Now one may think that the field in the pivot table should simply be renamed from "created_at" to "user_registered". This might work as long as you use this many-to-many relationship in this way:
`$user->articles()->get()`. However, as soon as you call the relationship from the article model (`$article->users()->get()`), Laravel will expect that the creation date field's name to be `created_at`.

I think the best solution would be to tell Laravel that `created_at` becomes a `user_registered` field only for the model for wich it was explicitly declared but NOT for fields in related pivot tables.

For now, I have just renamed the field in the pivot table to `user_registered` because in my case, I know that I won't need to call the relationship from the articles side `$article->users()->get()`. However, as soon as I need to call it this way, I wouldn't have a solution. Furthermore, I had to set the `updated_at` field appropriately in my User model: `const UPDATED_AT = 'updated_at'; `.
